client/client-lib/src/lib.rs

```diff
@@ -446,7 +448,7 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
                 },
                 &self.mint_client().config.tbs_pks,
             );
-            dbtx.commit_tx().expect("committing tx failed"); // FIXME: retry?
+            dbtx.commit_tx().await.expect("committing tx failed"); // FIXME: retry?
```

Comment:
We might need to wrap every `commit_tx` in a retry loop, because they can potentially fail due to concurrency, no?

Reply:
I think it's even a bit worse: we might need to re-run all the code since the tx started since it might have read now-outdated data and tries to update that (e.g. in case of the e-cash key counter).

Reply:
Yes, the loop needs to be over whole db transaction.

Reply:
Yes, agreed that the loop must be over the whole transaction because of the snapshot isolation. I can make this change.

@elsirion are you aware of any other areas where we might run into write-write conflicts?

Reply:
Not really, probably rather on the client since on the server side there's typically no parallelism. The client as it stands isn't thread safe at all and typically wrapped in a mutex. I hope DB tx can fix that.

Reply:
Ok I have added a loop that will try 5 times to commit the tx to the database before giving up and returning error, so that the introduction of the loop doesn't cause the command to hang forever (even if that's very unlikely). 5 was chosen arbitrarily as this issue should be rare for most users (need parallel clients trying to spend at the same time).

Reply:
Should we just make it `task_handle.is_shutting_down()`-based?

Reply:
I think this is a good idea but will leave it for another PR. Right now task_group.spawn swallows the return value making it difficult propagate the return value to the client.